### PR TITLE
Add doc about package references, korebuild, and an intro doc

### PR DESCRIPTION
**Pull request by natemcmaster**
Start adding some docs to explain what our build tools do

### CODE REVIEW

File: docs/KoreBuild.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+KoreBuild
+=========
+
+KoreBuild is a set of MSBuild targets and tasks used to define a common set of build and test tasks. The entry point for KoreBuild is defined in [KoreBuild.proj][korebuild-proj].
+
+## KoreBuild Lifecycle
+
+The KoreBuild chains together these targets. Custom targets can chain of these
```
Comment (ryanbrandenburg) [inline docs/KoreBuild.md]: "chain off these."

File: docs/KoreBuild.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+1. Prepare - pre-build actions like making directories
+1. Restore - NuGet restore
+1. Compile - calls /t:Build on \*.sln files
+1. Package - NuGet pack and other packaging steps
+1. Test - invokes VSTest
+1. Verify - post build tests
+1. Build = the default target. Runs Prepare;Restore;Compile;Package;Test;Verify
```
Comment (ryanbrandenburg) [inline docs/KoreBuild.md]: This format implies that Build will be run after Verify. If you're just listing them and not trying to imply an order of execution I would suggest a bullet-ed list.

File: docs/KoreBuild.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+## Extensibility points
+
+### Modules
+
+KoreBuild is designed to be a modular system. It is written as a backbone of default lifecycle targets and imports.
+Default functionality, such as building solutions and testing with VSTest, are written as modules in `files/KoreBuild/modules`.
+
+Other default functionality that require tasks are built from `modules/` in this repo. These include tasks for download NuGet packages,
```
Comment (ryanbrandenburg) [inline docs/KoreBuild.md]: "tasks for downloading". Also why include a newline here?

Comment (natemcmaster) [reply docs/KoreBuild.md]: thanks. update

File: docs/KoreBuild.md (hunk 2 above)
Comment (natemcmaster) [reply docs/KoreBuild.md]: updated

**Pull request merged**